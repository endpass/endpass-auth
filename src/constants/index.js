export const METHODS = {
  SIGN: 'SIGN',
  ACCOUNT: 'ACCOUNT',
  RECOVER: 'RECOVER',
  GET_SETTINGS: 'GET_SETTINGS',

  AUTH: 'AUTH',
  LOGOUT: 'LOGOUT',
  DIALOG_RESIZE: 'DIALOG_RESIZE',
  DIALOG_OPEN: 'DIALOG_OPEN',
  DIALOG_CLOSE: 'DIALOG_CLOSE',
  INITIATE: 'INITIATE',
  READY_STATE_BRIDGE: 'READY_STATE_BRIDGE',
};

export const LS_SETTINGS = 'endpass-connect:settings';

export const IDENTITY_MODE = Object.freeze({
  DEFAULT: 'default',
  CUSTOM: 'custom',
  LOCAL: 'local',
});

export const DIRECTION = Object.freeze({
  AUTH: 'auth',
  CONNECT: 'connect',
});

export const ORIGIN_HOST = (() => {
  const parser = document.createElement('a');
  parser.href = document.referrer;
  return parser.origin;
})();
