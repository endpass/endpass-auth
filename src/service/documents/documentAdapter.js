export default function documentAdapter(item) {
  return {
    ...item,
    imgPath: `${ENV.VUE_APP_IDENTITY_API_URL}/documents/${item.id}/front/file`,
    imgPathBack: `${ENV.VUE_APP_IDENTITY_API_URL}/documents/${item.id}/back/file`,
  };
}
