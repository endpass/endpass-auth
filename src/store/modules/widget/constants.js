export const WIDGET_RESIZE_DURATION = 300;

export default {};
