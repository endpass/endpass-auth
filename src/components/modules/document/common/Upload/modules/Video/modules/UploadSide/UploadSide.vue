<template>
  <div>
    <upload-header :document-type="documentType" />
    <drop-area
      :is-loading="isLoading"
      :error.sync="error"
      :description-supported-files="
        $t('components.uploadVideo.descriptions.supportedFiles')
      "
      :accept-files="$options.ACCEPT_VIDEO"
      @change="onFileChange"
    >
      <document-upload-front
        :error="error"
        :is-loading="isLoading"
        :progress-value="$options.uploadSideController.progress"
        :progress-label="$options.uploadSideController.progressLabel"
        :file="selectedFile"
        @file-remove="onFileRemove"
      />
    </drop-area>
    <div>
      <v-button
        skin="quaternary"
        data-test="record-button"
        @click="onRecord"
      >
        {{ recordButtonTitle }}
      </v-button>
      <div class="upload-side-recorded-title">
        {{ recordStateTitle }}
      </div>
      <v-button
        :disabled="!uploadFile"
        :skin="uploadButtonSkin"
        data-test="upload-button"
        @click="onUpload"
      >
        {{ $t('components.uploadDocument.upload') }}
      </v-button>
    </div>
  </div>
</template>

<script>
import VButton from '@endpass/ui/kit/VButton';
import DocumentUploadFront from '@/components/forms/DocumentUploadForm/DocumentUploadFront';
import createUploadSideController from './UploadSide.controller';
import DropArea from '@/components/modules/document/common/Upload/common/DropArea';
import UploadHeader from '@/components/modules/document/common/Upload/common/UploadHeader';
import { ACCEPT_VIDEO } from '@/components/modules/document/common/Upload/upload.constants';

export default {
  name: 'UploadSide',

  ACCEPT_VIDEO,

  uploadSideController: createUploadSideController(),

  inject: ['$validator'],

  props: {
    documentType: {
      type: String,
      default: '',
    },

    recordedFile: {
      type: File,
      default: null,
    },
  },

  data: () => ({
    documentId: '',
    error: null,
    selectedFile: null,
    isLoading: false,
    isRecognitionError: false,
    isUploaded: false,
  }),

  computed: {
    uploadButtonSkin() {
      return this.uploadFile ? 'primary' : 'quaternary';
    },

    isUploadReady() {
      return !this.isLoading && !!this.selectedFile && !this.error;
    },

    recordButtonTitle() {
      return this.recordedFile
        ? this.$t('components.uploadVideo.choose.retake')
        : this.$t('components.uploadVideo.choose.record');
    },

    uploadFile() {
      return this.recordedFile || this.selectedFile;
    },

    recordStateTitle() {
      return this.recordedFile
        ? this.$t('components.uploadVideo.choose.recorded')
        : '';
    },
  },

  watch: {
    documentType() {
      this.error = '';
    },
  },

  methods: {
    onRecord() {
      this.$emit('next');
    },

    async onUpload() {
      await this.startCreateDocument();
      if (!this.documentId) {
        return;
      }
      await this.continueCreateDocument();
    },

    startUpload() {
      this.$emit('start-upload');
    },

    onFileRemove() {
      this.error = '';
      this.selectedFile = null;
      this.isRecognitionError = false;
    },

    onFileChange(files) {
      const [file] = files;
      this.selectedFile = file;
    },

    onCancel() {
      this.$emit('cancel');
    },

    async onRecognize() {
      try {
        this.isLoading = true;
        this.isRecognitionError = false;
        const document = await this.$options.uploadSideController.recognize(
          this.documentId,
        );
        this.handleConfirm(document);
      } catch (e) {
        this.isRecognitionError = true;
        this.error = e.message;
      } finally {
        this.isLoading = false;
      }
    },

    async startCreateDocument() {
      try {
        this.isLoading = true;

        this.startUpload();
        this.documentId = await this.$options.uploadSideController.startCreateDocument(
          {
            file: this.uploadFile,
            type: this.documentType,
          },
        );
      } catch (e) {
        this.error = e.message;
      } finally {
        this.isLoading = false;
      }
    },

    async continueCreateDocument() {
      try {
        this.isLoading = true;
        this.isRecognitionError = false;
        const document = await this.$options.uploadSideController.continueCreateDocument(
          this.documentId,
        );
        this.handleConfirm(document);
      } catch (e) {
        this.isRecognitionError = true;
        this.error = e.message;
      } finally {
        this.isLoading = false;
      }
    },

    handleConfirm(document) {
      this.$emit('confirm', document);
    },
  },

  mounted() {
    this.$options.uploadSideController.init();
  },

  components: {
    VButton,
    UploadHeader,
    DropArea,
    DocumentUploadFront,
  },
};
</script>

<style lang="postcss">
.upload-side-recorded-title {
  height: 40px;
  font-size: 12px;
  color: var(--endpass-ui-color-green-2);
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
