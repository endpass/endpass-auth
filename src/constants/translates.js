import TranslateObjectFactory from '@/class/TranslateObjectFactory';

import { DOC_TYPES, DOC_STATUSES, PUBLIC_SCOPES } from '@/constants';

const { $t, createGetters } = TranslateObjectFactory;

export const DOC_TYPES_TRANSLATES = createGetters({
  [DOC_TYPES.PASSPORT]: $t('constants.documentType.passport'),
  [DOC_TYPES.DRIVER_LICENSE]: $t('constants.documentType.driverLicense'),
  [DOC_TYPES.PROOF_OF_ADDRESS]: $t('constants.documentType.proofOfAddress'),
  [DOC_TYPES.ID_CARD]: $t('constants.documentType.idCard'),
  [DOC_TYPES.SELFIE]: $t('constants.documentType.selfie'),
});

export const DOC_STATUSES_TRANSLATES = createGetters({
  [DOC_STATUSES.DRAFT]: $t('constants.documentStatus.draft'),
  [DOC_STATUSES.RECOGNITION]: $t('constants.documentStatus.recognition'),
  [DOC_STATUSES.PENDING_REVIEW]: $t('constants.documentStatus.pendingReview'),
  [DOC_STATUSES.NOT_READABLE]: $t('constants.documentStatus.notReadable'),
  [DOC_STATUSES.NOT_VERIFIED]: $t('constants.documentStatus.notVerified'),
  [DOC_STATUSES.VERIFIED]: $t('constants.documentStatus.verified'),
});

export const SCOPES_TRANSLATES = createGetters({
  // User
  [PUBLIC_SCOPES.USER]: $t('components.scopes.user.title'),
  [PUBLIC_SCOPES.USER_EMAIL_READ]: $t('components.scopes.user.email'),
  [PUBLIC_SCOPES.USER_ADDRESS_READ]: $t('components.scopes.user.address'),
  [PUBLIC_SCOPES.USER_PHONE_READ]: $t('components.scopes.user.phone'),
  // ID card
  [PUBLIC_SCOPES.DOCUMENTS_ID_CARD]: $t('components.scopes.idCard.title'),
  [PUBLIC_SCOPES.DOCUMENTS_ID_CARD_IMAGE_READ]: $t(
    'components.scopes.idCard.image',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_ID_CARD_STATUS_READ]: $t(
    'components.scopes.idCard.status',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_ID_CARD_DATA_READ]: $t(
    'components.scopes.idCard.data',
  ),
  // Passport
  [PUBLIC_SCOPES.DOCUMENTS_PASSPORT]: $t('components.scopes.passport.title'),
  [PUBLIC_SCOPES.DOCUMENTS_PASSPORT_IMAGE_READ]: $t(
    'components.scopes.passport.image',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PASSPORT_STATUS_READ]: $t(
    'components.scopes.passport.status',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PASSPORT_DATA_READ]: $t(
    'components.scopes.passport.data',
  ),
  // Driver license
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE]: $t(
    'components.scopes.driverLicense.title',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE_IMAGE_READ]: $t(
    'components.scopes.driverLicense.image',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE_STATUS_READ]: $t(
    'components.scopes.driverLicense.status',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE_DATA_READ]: $t(
    'components.scopes.driverLicense.data',
  ),
  // Proof of address
  [PUBLIC_SCOPES.DOCUMENTS_PROOF_OF_ADDRESS]: $t(
    'components.scopes.proofOfAddress.title',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PROOF_OF_ADDRESS_IMAGE_READ]: $t(
    'components.scopes.proofOfAddress.image',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PROOF_OF_ADDRESS_STATUS_READ]: $t(
    'components.scopes.proofOfAddress.status',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE_DATA_READ]: $t(
    'components.scopes.proofOfAddress.data',
  ),
});

export const SCOPES_TOOLTIPS_TRANSLATES = createGetters({
  [PUBLIC_SCOPES.USER]: $t('components.consentProvider.tooltips.user'),
  [PUBLIC_SCOPES.DOCUMENTS_ID_CARD]: $t(
    'components.consentProvider.tooltips.idCard',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PASSPORT]: $t(
    'components.consentProvider.tooltips.passport',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_DRIVER_LICENSE]: $t(
    'components.consentProvider.tooltips.driverLicense',
  ),
  [PUBLIC_SCOPES.DOCUMENTS_PROOF_OF_ADDRESS]: $t(
    'components.consentProvider.tooltips.proofOfAddress',
  ),
});
