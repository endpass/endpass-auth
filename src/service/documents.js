// @ts-check
import omitBy from 'lodash.omitby';
import isNil from 'lodash.isnil';
import get from 'lodash/get';
import generators from '@endpass/utils/generators';
import request from '@/class/singleton/request';
import { DOC_STATUSES, DOCUMENT_SIDES, UPLOAD_STATUSES } from '@/constants';

const docBaseURL = `${ENV.VUE_APP_IDENTITY_API_URL}/documents`;

const CHECK_RECOGNIZE_TIMEOUT = 1000;

/**
 * @param {Promise} promise
 * @return {Promise<any>}
 */
const withSuccess = promise => {
  return promise.then(res => {
    if (res && !res.success) throw new Error(res.message);

    return res;
  });
};

/**
 * @typedef {import('axios').AxiosRequestConfig} AxiosRequestConfig
 * */

const documentsService = {
  /**
   * @param {File} file UserDocument file
   * @returns {Promise}
   */
  async checkFile(file) {
    return withSuccess(request.upload(`${docBaseURL}/file/check`, { file }));
  },

  /**
   * @param {object} fields UserDocument object for upload
   * @param {string} fields.type UserDocument type
   * @param {string} [fields.description] UserDocument description
   */
  async createDocument({ type, description }) {
    const { message: docId } = await withSuccess(
      request.post(docBaseURL, omitBy({ type, description }, isNil)),
    );

    if (!docId) {
      throw new Error('Bad document id');
    }

    return docId;
  },

  /**
   * @param {string} docId
   * @return {Promise<void>}
   */
  async confirmDocument(docId) {
    return withSuccess(request.post(`${docBaseURL}/${docId}/confirm`));
  },

  /**
   * @param {object} data
   * @param {File} data.file
   * @param {string} data.docId
   * @param {AxiosRequestConfig} config
   * @return {Promise<void>}
   */
  async uploadFrontFile({ file, docId }, config) {
    return withSuccess(
      request.upload(`${docBaseURL}/${docId}/front`, { file }, config),
    );
  },

  /**
   * @param {object} data
   * @param {File} data.file
   * @param {string} data.docId
   * @param {AxiosRequestConfig} config
   * @return {Promise<void>}
   */
  async uploadBackFile({ file, docId }, config) {
    return withSuccess(
      request.upload(`${docBaseURL}/${docId}/back`, { file }, config),
    );
  },

  /**
   * @param {string} id
   * @return {Promise<*>}
   */
  async getDocumentStatus(id) {
    const document = await request.get(`${docBaseURL}/${id}`);
    if (!document) {
      throw new Error('Recognize error');
    }

    return document.status;
  },

  /**
   * @param {string} id
   * @return {Promise<void>}
   */
  async waitDocumentVerified(id) {
    const startTime = Date.now();
    const timeoutMS = 30000;
    const statuses = [DOC_STATUSES.VERIFIED];

    // eslint-disable-next-line no-unused-vars
    for await (const index of generators.repeatWithInterval(
      CHECK_RECOGNIZE_TIMEOUT,
    )) {
      const status = await documentsService.getDocumentStatus(id);

      if (statuses.includes(status)) {
        break;
      }

      const spendTime = Date.now() - startTime;
      if (timeoutMS && spendTime > timeoutMS) {
        break;
      }
    }
  },

  /**
   * @param {string} id
   * @return {Promise<void>}
   */
  async waitDocumentFinishRecognition(id) {
    const statuses = [DOC_STATUSES.RECOGNITION];

    // eslint-disable-next-line no-unused-vars
    for await (const index of generators.repeatWithInterval(
      CHECK_RECOGNIZE_TIMEOUT,
    )) {
      const status = await documentsService.getDocumentStatus(id);

      if (!statuses.includes(status)) {
        break;
      }
    }
  },

  /**
   *
   * @template {typeof UPLOAD_STATUSES} T
   * @param {string} id
   * @return {Promise<T[keyof T]>}
   */
  async getDocumentUploadStatusById(id) {
    const data = await request.get(`${docBaseURL}/${id}/status/upload`);
    const frontSideStatus = get(data, `${DOCUMENT_SIDES.FRONT}.status`);
    const backSideStatus = get(data, `${DOCUMENT_SIDES.BACK}.status`);

    if (backSideStatus === UPLOAD_STATUSES.NO_CONTENT) {
      return frontSideStatus;
    }

    if (frontSideStatus === UPLOAD_STATUSES.UPLOADED) {
      return backSideStatus;
    }

    return frontSideStatus;
  },

  /**
   *
   * @param {string} docId
   * @return {Promise<void>}
   */
  async waitDocumentUpload(docId) {
    // eslint-disable-next-line no-unused-vars
    for await (const index of generators.repeatWithInterval(
      CHECK_RECOGNIZE_TIMEOUT,
    )) {
      const status = await documentsService.getDocumentUploadStatusById(docId);

      if (status === UPLOAD_STATUSES.ERRORED) {
        throw new Error('Recognize error');
      }

      if (status !== UPLOAD_STATUSES.PROCESSING) {
        break;
      }
    }
  },

  /**
   * @param {string} clientId
   * @return {Promise<any>}
   */
  async getRequiredDocumentsTypes(clientId) {
    return request.get(
      `${ENV.VUE_APP_IDENTITY_API_URL}/apps/${clientId}/documents/required`,
    );
  },

  /**
   * @return {Promise<any>}
   */
  async getDocumentsList() {
    const { items, total } = await request.get(docBaseURL);
    return {
      items,
      total,
    };
  },
};

export default documentsService;
