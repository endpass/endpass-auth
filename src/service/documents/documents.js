// @ts-check
import mapToQueryString from '@endpass/utils/mapToQueryString';
import request from '@/class/singleton/request';
import documentAdapter from './documentAdapter';
import { filterNullableParams, getFileFormData } from '@/services/shared/utils';

const baseURL = `${ENV.VUE_APP_IDENTITY_API_URL}/documents`;

export default {
  /**
   * @param {string} id
   * @return {Promise<UserDocument>}
   */
  async getDocument(id) {
    const res = await request.get(`${baseURL}/${id}`);
    return documentAdapter(res);
  },

  /**
   *
   * @param {string} id document id
   * @return {Promise<UserDocumentLog[]>}
   */
  async getDocumentStatusLog(id) {
    return request.get(`${baseURL}/${id}/status/log`);
  },

  /**
   * @param {object} params
   * @param {number} [params.offset]
   * @param {number} [params.limit]
   * @param {string} [params.status]
   * @return {Promise<{total: number, items: UserDocument[]}>}
   */
  async getDocumentList({ offset = 0, limit = 10, status }) {
    const path = mapToQueryString(
      baseURL,
      filterNullableParams({
        offset,
        limit,
        status,
      }),
    );

    const { items, total } = await request.get(path);

    return {
      items: items.map(documentAdapter),
      total,
    };
  },

  /**
   * @param {File} file UserDocument file
   * @returns {Promise}
   */
  async checkFile(file) {
    return request.upload(`${baseURL}/file/check`, getFileFormData(file));
  },

  /**
   * @param {object} fields UserDocument object for upload
   * @param {string} fields.type UserDocument type
   * @param {string} [fields.description] UserDocument description
   */
  async createDocument({ type, description }) {
    const { message: docId } = await request.post(
      baseURL,
      filterNullableParams({ type, description }),
    );

    if (!docId) {
      throw new Error('Bad document id');
    }

    return docId;
  },

  /**
   * @param {object} data
   * @param {File} data.file
   * @param {string} data.docId
   * @param {{onUploadProgress: function}} config
   * @return {Promise<void>}
   */
  async uploadFrontFile({ file, docId }, config) {
    return request.upload(
      `${baseURL}/${docId}/front`,
      getFileFormData(file),
      config,
    );
  },

  /**
   * @param {object} data
   * @param {File} data.file
   * @param {string} data.docId
   * @param {{onUploadProgress: function}} config
   * @return {Promise<void>}
   */
  async uploadBackFile({ file, docId }, config) {
    return request.upload(
      `${baseURL}/${docId}/back`,
      getFileFormData(file),
      config,
    );
  },

  /**
   * @param {string} id
   * @return {Promise<>}
   */
  async removeDocument(id) {
    return request.delete(`${baseURL}/${id}`);
  },

  /**
   *
   * @param {string} id
   * @return {Promise<import('axios').AxiosResponse>}
   */
  getDocumentsUploadStatusById(id) {
    return request.get(`${baseURL}/${id}/status/upload`);
  },

  getDocumentsUploadStatus() {
    return request.get(`${baseURL}/status/upload`);
  },
};
