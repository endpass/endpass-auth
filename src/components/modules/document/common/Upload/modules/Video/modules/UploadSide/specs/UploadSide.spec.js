import VeeValidate from 'vee-validate';
import UIComponents from '@endpass/ui';

import { shallowMount, createLocalVue } from '@vue/test-utils';
import { document } from '@unitFixtures/documents';
import setupI18n from '@/locales/i18nSetup';

import UploadSide from '../UploadSide';
import documentsService from '@/service/documents';
import riskScoringService from '@/service/riskScoring';
import { DOC_TYPES } from '@/constants';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);
localVue.use(VeeValidate);
localVue.use(UIComponents);

describe('UploadVideo > UploadSide', () => {
  let wrapper;

  const file = new File([''], 'filename');

  const createWrapper = options => {
    return shallowMount(UploadSide, {
      localVue,
      i18n,
      sync: false,
      propsData: {
        documentType: DOC_TYPES.SELFIE,
        recordedFile: file,
      },
      ...options,
    });
  };

  beforeEach(() => {
    jest.useFakeTimers();
    jest.clearAllMocks();
    wrapper = createWrapper();
  });

  describe('render', () => {
    it('should correctly render component', () => {
      expect.assertions(2);

      expect(wrapper.name()).toBe('UploadSide');
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should suggest to use mobile app if selfie', async () => {
      expect.assertions(1);

      expect(wrapper.find('mobile-suggestions-stub').exists()).toBe(true);
    });
  });

  describe('behavior', () => {
    it('should upload recorded file and confirm document', async () => {
      expect.assertions(2);

      expect(wrapper.emitted().confirm).toBeUndefined();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(wrapper.emitted().confirm).toEqual([[document]]);
    });

    it('should upload front side of document and recognize', async () => {
      expect.assertions(2);

      expect(wrapper.emitted().confirm).toBeUndefined();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(wrapper.emitted().confirm).toEqual([[document]]);
    });

    it('should send fingerprint after upload', async () => {
      expect.assertions(2);

      expect(riskScoringService.sendUserMetrics).not.toBeCalled();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(riskScoringService.sendUserMetrics).toBeCalledTimes(1);
    });

    it('should not emit confirm, if error in recognize', async () => {
      expect.assertions(2);

      documentsService.confirmDocument.mockRejectedValueOnce(new Error());

      expect(wrapper.emitted().confirm).toBeUndefined();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(wrapper.emitted().confirm).toBeUndefined();
    });

    it('should not emit confirm, if error when upload', async () => {
      expect.assertions(1);

      documentsService.uploadFrontFile.mockRejectedValueOnce(new Error());

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(wrapper.emitted().confirm).toBeUndefined();
    });

    it('should show error, if error when upload', async () => {
      expect.assertions(1);

      documentsService.uploadFrontFile.mockRejectedValueOnce(new Error());

      wrapper.find('[data-test=upload-button]').vm.$emit('click');

      await global.flushPromises();
      await wrapper.vm.$nextTick();

      expect(wrapper.find('[data-test=upload-side-error]').text()).toMatch(
        wrapper.vm.$t('store.error.uploadDocument.default'),
      );
    });

    it('should emit confirm after recognize', async () => {
      expect.assertions(2);

      expect(wrapper.emitted().confirm).toBeUndefined();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(wrapper.emitted().confirm).toEqual([[document]]);
    });

    it('should confirm after upload', async () => {
      expect.assertions(2);

      expect(documentsService.confirmDocument).not.toBeCalled();

      wrapper.find('[data-test=upload-button]').vm.$emit('click');
      await global.flushPromises();

      expect(documentsService.confirmDocument).toBeCalledTimes(1);
    });
  });
});
