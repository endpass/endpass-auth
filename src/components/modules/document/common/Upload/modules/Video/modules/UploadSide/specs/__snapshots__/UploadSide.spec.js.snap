// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadVideo > UploadSide render should correctly render component 1`] = `
<div>
  <upload-header-stub documenttype="Passport"></upload-header-stub>
  <drop-area-stub descriptionsupportedfiles="Only .avi, .mpeg, .mp4 files" descriptionfilesize="" validateextensions="avi,mp4,mpeg" acceptmimetypes="video/avi,video/mpeg,video/mp4">
    <document-upload-front-stub progress-value="0" progress-label=""></document-upload-front-stub>
  </drop-area-stub>
  <!---->
  <div>
    <v-button-stub skin="quaternary" size="normal" data-test="record-button">
      Record selfie video
    </v-button-stub>
    <div class="upload-side-recorded-title">

    </div>
    <v-button-stub skin="quaternary" size="normal" disabled="true" data-test="upload-button">
      Upload
    </v-button-stub>
  </div>
</div>
`;
